behat/symfony2-extension/2.1/features/bootstrap/FeatureContext.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+use Behat\Behat\Context\Context;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\KernelInterface;
+
+/**
+ * This context class contains the definitions of the steps used by the demo 
+ * feature file. Learn how to get started with Behat and BDD on Behat's website.
+ * 
+ * @see http://behat.org/en/latest/quick_start.html
+ */
+class FeatureContext implements Context
+{
+    /**
+     * @var KernelInterface
+     */
+    private $kernel;
+
+    /**
+     * @var Response|null
+     */
+    private $response;
+
+    public function __construct(KernelInterface $kernel)
+    {
+        $this->kernel = $kernel;
+    }
+
+    /**
+     * @When a demo scenario sends a request to :path
+     */
+    public function aDemoScenarioSendsARequestTo(string $path)
+    {
+        $this->response = $this->kernel->handle(Request::create($path, 'GET'));
+    }
+
+    /**
+     * @Then the response should be received
+     */
+    public function theResponseShouldBeReceived()
+    {
+        assert($this->response !== null);
```

Comment:
`assert` is not a good implementation here. This may not always run, and it may not always throw an exception on failures (it does not throw by default).

Behat wants you to throw an exception on failure in step definitions.